src/riak_core_job_manager.erl

```diff
@@ -0,0 +1,1310 @@
+%% A single statistic.
+%% <i>Most</i> Service statistics are integral counters in the form
+%% <code>{<i>Key</i>, <i>Count</i> :: non_neg_integer()}</code>.
+
+-type stat_key() :: atom() | tuple().
+%% The Key by which a statistic is referenced.
+%% <i>Most</i> statistics keys are simple, single-word atoms.
+
+-type stat_val() :: term().
+%% The value of a statistic, most often an integral count greater than zero.
+
+%% ===================================================================
+%% Public API
+%% ===================================================================
+
+-spec cancel(
+    JobOrId :: riak_core_job:job() | riak_core_job:gid(),
+    Kill    :: boolean())
+        ->  {ok, killed | canceled | history, riak_core_job:job()}
+            | {error, running, riak_core_job:job()} | false | {error, term()}.
+%%
+%% @doc Cancel, or optionally kill, the specified Job.
+%%
+%% If <i>Kill</i> is `true' the job will de-queued or killed, as appropriate.
+%% If <i>Kill</i> is `false' the job will only be de-queued.
+%%
+%% Returns:
+%% <dl>
+%%  <dt><code>{ok, <i>Status</i>, <i>Job</i>}</code></dt>
+%%  <dd>Status indicates whether the Job was `killed', `canceled' (de-queued),
+%%      or had recently `finished' running.</dd>
+%%  <dd>Note that what constitutes "recently finished" is subject to load and
+%%      history configuration.</dd>
+%%  <dd><i>Job</i> is the current instance of <i>JobOrId</i>.</dd>
+%%
+%%  <dt><code>{error, running, <i>Job</i>}</code></dt>
+%%  <dd>Only returned when <i>Kill</i> is `false', indicating that <i>Job</i>
+%%      is currently active (and remains so).</dd>
+%%  <dd><i>Job</i> is the current instance of <i>JobOrId</i>.</dd>
+%%
+%%  <dt><code>false</code></dt>
+%%  <dd><i>JobOrId</i> is not, or is no longer, known to the Manager.</dd>
+%% </dl>
+%%
+cancel(JobOrId, Kill) ->
+    JobId = case riak_core_job:version(JobOrId) of
+        {job, _} ->
+            riak_core_job:gid(JobOrId);
+        _ ->
+            JobOrId
+    end,
+    gen_server:call(?JOBS_MGR_NAME, {cancel, JobId, Kill}).
+
+-spec config() -> config() | {error, term()}.
+%%
+%% @doc Return the current effective Jobs Management configuration.
+%%
+config() ->
+    gen_server:call(?JOBS_MGR_NAME, config).
+
+-spec find(JobOrId :: riak_core_job:job() | riak_core_job:gid())
+        -> {ok, riak_core_job:job()} | false | {error, term()}.
+%%
+%% @doc Return the latest instance of the specified Job.
+%%
+%% `false' is returned if the Job is not, or is no longer, known to the Manager.
+%%
+find(JobOrId) ->
+    JobId = case riak_core_job:version(JobOrId) of
+        {job, _} ->
+            riak_core_job:gid(JobOrId);
+        _ ->
+            JobOrId
+    end,
+    gen_server:call(?JOBS_MGR_NAME, {find, JobId}).
+
+-spec reconfigure() -> ok.
+%%
+%% @doc Re-read application environment configuration and adjust accordingly.
+%%
+reconfigure() ->
+    gen_server:cast(?JOBS_MGR_NAME, ?job_svc_cfg_token).
```

Comment:
I read the comments where this token is defined, but this still feels a bit odd and unidiomatic to me. Why are we using a special token here, and not just an atom like `reload_job_config` or something? In theory, I guess it's possible that someone else might write code somewhere else in Riak that uses the same atom and then somehow accidentally sends it to the wrong process, but that seems exceedingly unlikely in practice.